modules/openapi-generator/src/main/java/org/openapitools/codegen/utils/ModelUtils.java

```diff
@@ -1255,18 +1254,18 @@ public static String getParentName(ComposedSchema composedSchema, Map<String, Sc
                         }
                     } else {
                         // not a parent since discriminator.propertyName is not set
-                        refedWithoutDiscriminator.add(parentName);
                     }
                 } else {
                     // not a ref, doing nothing
                 }
             }
         }
 
-        if (names.size() == 0 && refedWithoutDiscriminator.size() == 1) {
-            LOGGER.warn("[deprecated] inheritance without use of 'discriminator.propertyName' is deprecated " +
```

Comment:
Doesn't the spec (at least 3.0.3 does as far as I can tell) require a `discriminator`?

Reply:
It definitely does.
I have no idea why this "deprecated" case is handled this way. What this PR does is making sure both `parent` and `allParents` consistently handle it the same way.

We might want to remove this deprecated logic in generator 5.0

Reply:
In that case, wouldn't it make sense to leave the warning in until the (language-agnostic part of the) generator just flat out rejects specs with missing discriminators? 
Or does it already do that?

Reply:
This warning still lives in `getParentName()`, where it was originally introduced.
In my earlier PR I duplicated the logic from `getParentName()` into this `getAllParentsName()` - together with the warning. Apparently it was not a robust approach, so i this PR I remove the duplicated logic, and just call `getParentName` from `getAllParentsName`.
This way `getAllParentsName` doesn't really care about any special handling in `getParentName`

Reply:
the "deprecated" behavior is to support openapi 2.0

Reply:
Alright, if it's still handled in `getParentName()` that lgtm